Panic/crash on "unknown cert key type ssh-rsa-cert-v01@openssh.com"

We just started using the SSH CA backend, and today Vault crashed on me with this:

```
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: panic: unknown cert key type ssh-rsa-cert-v01@openssh.com
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: goroutine 1188208 [running]:
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).Type(0xc4205766e8, 0xc422947400, 0x48d)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:480 +0x124
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).Marshal(0xc4205766e8, 0xc4230b0fa0, 0xc4205767d0, 0xc420576740)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:468 +0x34e
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).bytesForSigning(0xc422321290, 0x2696bc0, 0xc423912480, 0x20)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:438 +0x6b
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).SignCert(0xc422321290, 0x2686ac0, 0xc42000e240, 0x268ffc0, 0xc4230b0e80
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:407 +0x104
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/builtin/logical/ssh.(*creationBundle).sign(0xc420576b48, 0x68f, 0x700, 0x268ffc0)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:408 +0x2b1
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/builtin/logical/ssh.(*backend).pathSignCertificate(0xc4204e0a00, 0xc421529500, 0xc420b25a50, 0xc422ca0a20, 0x7, 0xc42
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:177 +0x10e4
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/builtin/logical/ssh.(*backend).pathSign(0xc4204e0a00, 0xc421529500, 0xc420b25a50, 0x5010104, 0x0, 0xffffffffffffffff)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:98 +0x31b
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/builtin/logical/ssh.(*backend).(github.com/hashicorp/vault/builtin/logical/ssh.pathSign)-fm(0xc421529500, 0xc420b25a5
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:38 +0x3e
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/logical/framework.(*Backend).HandleRequest(0xc4209fc8f0, 0xc421529500, 0xc421529500, 0xc423129664, 0x10)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/logical/framework/backend.go:221 +0x4c8
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vault.(*Router).routeCommon(0xc42000e870, 0xc421529500, 0x0, 0x0, 0x5000000, 0x0, 0x0)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/router.go:326 +0x636
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vault.(*Router).Route(0xc42000e870, 0xc421529500, 0xc421529500, 0xc420e80240, 0x0)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/router.go:218 +0x3a
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vault.(*Core).handleRequest(0xc4201b7c00, 0xc421529500, 0x0, 0x0, 0x0, 0x0)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_handling.go:188 +0xb10
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vault.(*Core).HandleRequest(0xc4201b7c00, 0xc421529500, 0x0, 0x0, 0x0)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_handling.go:45 +0xc77
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/http.request(0xc4201b7c00, 0x2693e40, 0xc4231298a0, 0xc425391900, 0xc421529500, 0x0, 0x0)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/handler.go:209 +0x3c
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/http.handleLogical.func1(0x2693e40, 0xc4231298a0, 0xc425391900)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/logical.go:121 +0xfb
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: net/http.HandlerFunc.ServeHTTP(0xc4203f25c0, 0x2693e40, 0xc4231298a0, 0xc425391900)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/http.handleRequestForwarding.func1(0x2693e40, 0xc4231298a0, 0xc425391900)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/handler.go:168 +0x761
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: net/http.HandlerFunc.ServeHTTP(0xc4203f25e0, 0x2693e40, 0xc4231298a0, 0xc425391900)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: net/http.(*ServeMux).ServeHTTP(0xc4203a40f0, 0x2693e40, 0xc4231298a0, 0xc425391900)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:2238 +0x130
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/http.wrapHelpHandler.func1(0x2693e40, 0xc4231298a0, 0xc425391900)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/help.go:22 +0x17f
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: net/http.HandlerFunc.ServeHTTP(0xc4203f2620, 0x2693e40, 0xc4231298a0, 0xc425391900)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/http.wrapGenericHandler.func1(0x2693e40, 0xc4231298a0, 0xc425391900)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/handler.go:86 +0xb1
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: net/http.HandlerFunc.ServeHTTP(0xc4203f2640, 0x2693e40, 0xc4231298a0, 0xc425391900)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vault.(*forwardedRequestRPCServer).ForwardRequest(0xc420e68480, 0x7f511f6e9f90, 0xc424c054a0, 0xc42043b3b0, 0xc420e68
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_forwarding.go:355 +0x149
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vault._RequestForwarding_ForwardRequest_Handler(0x180aea0, 0xc420e68480, 0x7f511f6e9f90, 0xc424c054a0, 0xc421532280, 
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_forwarding_service.pb.go:148 +0x28d
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc4209c4140, 0x269d760, 0xc4221ff5c0, 0xc425391800, 0xc420e8
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/google.golang.org/grpc/server.go:791 +0xc41
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vendor/google.golang.org/grpc.(*Server).handleStream(0xc4209c4140, 0x269d760, 0xc4221ff5c0, 0xc425391800, 0x0)
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/google.golang.org/grpc/server.go:991 +0x15a6
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: github.com/hashicorp/vault/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4213e44b0, 0xc4209c4140, 0x269d760, 0xc4221ff5c0, 0xc
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/google.golang.org/grpc/server.go:561 +0xa9
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]: created by github.com/hashicorp/vault/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
Jun 15 17:48:57 ip-10-183-2-238.ec2.internal bash[1873]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/google.golang.org/grpc/server.go:562 +0xa1
```

I can see where in https://github.com/golang/crypto/blob/master/ssh/certs.go it throws this message with a `panic`.  Without even worrying about how a user ended up passing in a cert that triggered that or considering if that should be a supported key type, my top concern is why it crashes the Vault instance completely.  Seems like it should be caught and reported as an error without any impact on the process.

If useful, stack traces:

Forwarding:
```
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: 2017/06/20 23:28:46.339133 [ERROR] forwarding: panic serving request for %v: %v
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: %s: /v1/ssh/sign/monuser="unknown cert key type ssh-rsa-cert-v01@openssh.com" goroutine 1715 [running]:
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*forwardedRequestRPCServer).ForwardRequest.func1.1(0xc4213ae7e0, 0xc4207a3ae0, 0xc42156fa00, 0xc42064d867)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_forwarding.go:365 +0xbc
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: panic(0x171ad60, 0xc4210d9190)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/runtime/panic.go:489 +0x2cf
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).Type(0xc42064c550, 0xc4215fa000, 0x48d)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:480 +0x124
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).Marshal(0xc42064c550, 0xc4210cd4e0, 0xc42064c638, 0xc42064c5a8)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:468 +0x34e
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).bytesForSigning(0xc4201f00b0, 0x26eec80, 0xc421385620, 0x20)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:438 +0x6b
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).SignCert(0xc4201f00b0, 0x26deb40, 0xc42000e240, 0x26e8080, 0xc4210cd44
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:407 +0x104
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/builtin/logical/ssh.(*creationBundle).sign(0xc42064c9b0, 0xcaf, 0xd80, 0x26e8080)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:408 +0x2b1
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/builtin/logical/ssh.(*backend).pathSignCertificate(0xc42040c880, 0xc4210e36c0, 0xc4210d8660, 0xc4213b8000, 0x7, 0xc4
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:177 +0x10e4
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/builtin/logical/ssh.(*backend).pathSign(0xc42040c880, 0xc4210e36c0, 0xc4210d8660, 0x7010103, 0x0, 0xffffffffffffffff
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:98 +0x31b
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/builtin/logical/ssh.(*backend).(github.com/hashicorp/vault/builtin/logical/ssh.pathSign)-fm(0xc4210e36c0, 0xc4210d86
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:38 +0x3e
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/logical/framework.(*Backend).HandleRequest(0xc420427a00, 0xc4210e36c0, 0xc4210e36c0, 0xc4210b77c4, 0x10)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/logical/framework/backend.go:221 +0x4c8
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*Router).routeCommon(0xc4201bff40, 0xc4210e36c0, 0x0, 0x0, 0x7000000, 0x0, 0x0)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/router.go:348 +0x636
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*Router).Route(0xc4201bff40, 0xc4210e36c0, 0xc4210e36c0, 0xc420784de0, 0x0)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/router.go:240 +0x3a
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*Core).handleRequest(0xc4201f9c00, 0xc4210e36c0, 0x0, 0x0, 0x0, 0x0)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_handling.go:188 +0xb10
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*Core).HandleRequest(0xc4201f9c00, 0xc4210e36c0, 0x0, 0x0, 0x0)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_handling.go:45 +0xc77
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.request(0xc4201f9c00, 0x26ebf00, 0xc4210b78e0, 0xc42156fa00, 0xc4210e36c0, 0x0, 0x0)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/handler.go:210 +0x3c
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.handleLogical.func1(0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/logical.go:122 +0xfb
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebd40, 0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.handleRequestForwarding.func1(0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/handler.go:169 +0x761
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebd60, 0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: net/http.(*ServeMux).ServeHTTP(0xc42016b470, 0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:2238 +0x130
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.wrapHelpHandler.func1(0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/help.go:22 +0x17f
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebda0, 0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.wrapCORSHandler.func1(0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/cors.go:37 +0x5a0
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebdc0, 0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.wrapGenericHandler.func1(0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/handler.go:87 +0xb1
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebde0, 0x26ebf00, 0xc4210b78e0, 0xc42156fa00)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*forwardedRequestRPCServer).ForwardRequest.func1()
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_forwarding.go:371 +0xab
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*forwardedRequestRPCServer).ForwardRequest(0xc4207a3ae0, 0x7f91dc9ae9f8, 0xc4213ae630, 0xc4205182a0, 0xc4207a
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_forwarding.go:373 +0x1b2
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault._RequestForwarding_ForwardRequest_Handler(0x1839320, 0xc4207a3ae0, 0x7f91dc9ae9f8, 0xc4213ae630, 0xc4213506e0,
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_forwarding_service.pb.go:148 +0x28d
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc420642140, 0x26f58e0, 0xc42123d1a0, 0xc42156f900, 0xc4207
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/google.golang.org/grpc/server.go:776 +0xc41
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/google.golang.org/grpc.(*Server).handleStream(0xc420642140, 0x26f58e0, 0xc42123d1a0, 0xc42156f900, 0x0)
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/google.golang.org/grpc/server.go:976 +0x15a6
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4210c8ac0, 0xc420642140, 0x26f58e0, 0xc42123d1a0, 0x
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/google.golang.org/grpc/server.go:546 +0xa9
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: created by github.com/hashicorp/vault/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/google.golang.org/grpc/server.go:547 +0xa1
Jun 20 23:28:46 ip-10-183-0-128.ec2.internal bash[31671]: =[unknown!]
```

HTTP:
```
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: 2017/06/20 23:28:45 http2: panic serving 10.183.3.182:42414: unknown cert key type ssh-rsa-cert-v01@openssh.com
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: goroutine 1698 [running]:
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/net/http2.(*serverConn).runHandler.func1(0xc42000d6b0, 0xc42064dfaf, 0xc421426380)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/net/http2/server.go:2046 +0x190
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: panic(0x171ad60, 0xc4210d1460)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/runtime/panic.go:489 +0x2cf
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).Type(0xc42064cad0, 0xc42143af00, 0x48d)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:480 +0x124
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).Marshal(0xc42064cad0, 0xc4210baf00, 0xc42064cbb8, 0xc42064cb28)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:468 +0x34e
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).bytesForSigning(0xc4214c24d0, 0x26eec80, 0xc42122fda0, 0x20)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:438 +0x6b
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh.(*Certificate).SignCert(0xc4214c24d0, 0x26deb40, 0xc42000e240, 0x26e8080, 0xc4210bae8
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/crypto/ssh/certs.go:407 +0x104
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/builtin/logical/ssh.(*creationBundle).sign(0xc42064cf30, 0xcaf, 0xd80, 0x26e8080)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:408 +0x2b1
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/builtin/logical/ssh.(*backend).pathSignCertificate(0xc42040c880, 0xc4210e3340, 0xc4210d0950, 0xc4212a0240, 0x7, 0xc4
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:177 +0x10e4
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/builtin/logical/ssh.(*backend).pathSign(0xc42040c880, 0xc4210e3340, 0xc4210d0950, 0x7010102, 0x0, 0xffffffffffffffff
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:98 +0x31b
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/builtin/logical/ssh.(*backend).(github.com/hashicorp/vault/builtin/logical/ssh.pathSign)-fm(0xc4210e3340, 0xc4210d09
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/builtin/logical/ssh/path_sign.go:38 +0x3e
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/logical/framework.(*Backend).HandleRequest(0xc420427a00, 0xc4210e3340, 0xc4210e3340, 0xc4210ab744, 0x10)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/logical/framework/backend.go:221 +0x4c8
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*Router).routeCommon(0xc4201bff40, 0xc4210e3340, 0x0, 0x0, 0x6000000, 0x0, 0x0)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/router.go:348 +0x636
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*Router).Route(0xc4201bff40, 0xc4210e3340, 0xc4210e3340, 0xc420784de0, 0x0)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/router.go:240 +0x3a
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*Core).handleRequest(0xc4201f9c00, 0xc4210e3340, 0x0, 0x0, 0x0, 0x0)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_handling.go:188 +0xb10
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vault.(*Core).HandleRequest(0xc4201f9c00, 0xc4210e3340, 0x0, 0x0, 0x0)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vault/request_handling.go:45 +0xc77
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.request(0xc4201f9c00, 0x26eedc0, 0xc42000d6b0, 0xc421288400, 0xc4210e3340, 0x0, 0x0)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/handler.go:210 +0x3c
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.handleLogical.func1(0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/logical.go:122 +0xfb
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebd40, 0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.handleRequestForwarding.func1(0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/handler.go:169 +0x761
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebd60, 0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.(*ServeMux).ServeHTTP(0xc42016b470, 0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:2238 +0x130
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.wrapHelpHandler.func1(0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/help.go:22 +0x17f
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebda0, 0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.wrapCORSHandler.func1(0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/cors.go:37 +0x5a0
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebdc0, 0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/http.wrapGenericHandler.func1(0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/http/handler.go:87 +0xb1
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.HandlerFunc.ServeHTTP(0xc4201ebde0, 0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:1942 +0x44
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.serverHandler.ServeHTTP(0xc420169ef0, 0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:2568 +0x92
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.initNPNRequest.ServeHTTP(0xc4205ffc00, 0xc420169ef0, 0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/server.go:3088 +0x93
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.(*initNPNRequest).ServeHTTP(0xc4210a9b90, 0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         <autogenerated>:312 +0x74
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: net/http.(Handler).ServeHTTP-fm(0x26eedc0, 0xc42000d6b0, 0xc421288400)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /usr/local/Cellar/go/1.8.3/libexec/src/net/http/h2_bundle.go:4331 +0x4d
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: github.com/hashicorp/vault/vendor/golang.org/x/net/http2.(*serverConn).runHandler(0xc421426380, 0xc42000d6b0, 0xc421288400, 0xc4210ab780)
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/net/http2/server.go:2053 +0x89
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]: created by github.com/hashicorp/vault/vendor/golang.org/x/net/http2.(*serverConn).processHeaders
Jun 20 23:28:45 ip-10-183-0-128.ec2.internal bash[31671]:         /Users/bkrodg/go/src/github.com/hashicorp/vault/vendor/golang.org/x/net/http2/server.go:1787 +0x4ac
```

--- a/builtin/logical/ssh/path_sign.go
+++ b/builtin/logical/ssh/path_sign.go
@@ -389,6 +389,16 @@ func (b *backend) calculateTTL(data *framework.FieldData, role *sshRole) (time.D
 }
 
 func (b *creationBundle) sign() (retCert *ssh.Certificate, retErr error) {
+	defer func() {
+		if r := recover(); r != nil {
+			errMsg, ok := r.(string)
+			if ok {
+				retCert = nil
+				retErr = errors.New(errMsg)
+			}
+		}
+	}()
+
 	serialNumber, err := certutil.GenerateSerialNumber()
 	if err != nil {
 		return nil, err

--- a/builtin/logical/ssh/path_sign.go
+++ b/builtin/logical/ssh/path_sign.go
@@ -389,16 +389,6 @@ func (b *backend) calculateTTL(data *framework.FieldData, role *sshRole) (time.D
 }
 
 func (b *creationBundle) sign() (retCert *ssh.Certificate, retErr error) {
-	defer func() {
-		if r := recover(); r != nil {
-			err, ok := r.(error)
-			if ok {
-				retCert = nil
-				retErr = err
-			}
-		}
-	}()
-
 	serialNumber, err := certutil.GenerateSerialNumber()
 	if err != nil {
 		return nil, err

--- a/vault/request_forwarding.go
+++ b/vault/request_forwarding.go
@@ -7,6 +7,7 @@ import (
 	"net"
 	"net/http"
 	"net/url"
+	"runtime"
 	"sync"
 	"sync/atomic"
 	"time"
@@ -352,11 +353,27 @@ func (s *forwardedRequestRPCServer) ForwardRequest(ctx context.Context, freq *fo
 	// meets the interface requirements.
 	w := forwarding.NewRPCResponseWriter()
 
-	s.handler.ServeHTTP(w, req)
+	resp := &forwarding.Response{}
+	var respSet bool
 
-	resp := &forwarding.Response{
-		StatusCode: uint32(w.StatusCode()),
-		Body:       w.Body().Bytes(),
+	runRequest := func() {
+		defer func() {
+			// Logic here comes mostly from the Go source code
+			if err := recover(); err != nil {
+				const size = 64 << 10
+				buf := make([]byte, size)
+				buf = buf[:runtime.Stack(buf, false)]
+				resp.StatusCode = 500
+				s.core.logger.Error("forwarding: panic serving request for %v: %v\n%s", req.URL.Path, err, buf)
+				respSet = true
+			}
+		}()
+		s.handler.ServeHTTP(w, req)
+	}
+	runRequest()
+	if !respSet {
+		resp.StatusCode = uint32(w.StatusCode())
+		resp.Body = w.Body().Bytes()
 	}
 
 	header := w.Header()

--- a/vault/request_forwarding.go
+++ b/vault/request_forwarding.go
@@ -354,7 +354,6 @@ func (s *forwardedRequestRPCServer) ForwardRequest(ctx context.Context, freq *fo
 	w := forwarding.NewRPCResponseWriter()
 
 	resp := &forwarding.Response{}
-	var respSet bool
 
 	runRequest := func() {
 		defer func() {
@@ -363,18 +362,14 @@ func (s *forwardedRequestRPCServer) ForwardRequest(ctx context.Context, freq *fo
 				const size = 64 << 10
 				buf := make([]byte, size)
 				buf = buf[:runtime.Stack(buf, false)]
-				resp.StatusCode = 500
-				s.core.logger.Error("forwarding: panic serving request for %v: %v\n%s", req.URL.Path, err, buf)
-				respSet = true
+				s.core.logger.Error("forwarding: panic serving request", "path", req.URL.Path, "error", err, "stacktrace", buf)
 			}
 		}()
 		s.handler.ServeHTTP(w, req)
 	}
 	runRequest()
-	if !respSet {
-		resp.StatusCode = uint32(w.StatusCode())
-		resp.Body = w.Body().Bytes()
-	}
+	resp.StatusCode = uint32(w.StatusCode())
+	resp.Body = w.Body().Bytes()
 
 	header := w.Header()
 	if header != nil {

